### PR TITLE
Allow users to see merged stories they submitted

### LINKED ISSUE #423: Merged submissions disappear from a user's submission list
A story I submitted was merged into another story. My submission is missing from my own submission list. I would expect it to be there, maybe with some indication that it has been merged into another, but, when clicked, redirect to the merged story when visiting it.

Reproduction:

1. When I go to https://lobste.rs/newest/colindean
2. I expect to see https://lobste.rs/s/xun2wv/severe_flaw_wpa2_protocol_leaves_wi_fi at the top of the list
    * this story has been merged with https://lobste.rs/s/dwzplh/krack_attacks_breaking_wpa2
3. I do not see this story there.

### PR DESCRIPTION
**Pull request by jmks**
Fixes #423 - Users can now see their own stories that were merged into another story.

The logic for finding all those stories was slightly more complicated, so I pushed it into the `StoryRepository` so that it would be easier to test.

For the view, I reused the existing partial (`stories/listdetail` is used in many places). When the submitter is viewing their own stories, a merged story looks the same as if they were seeing the story view.

I am pretty sure this will conflict with my other PR #563, so once one is merged, I can update the other.

### CODE REVIEW

**Comment by pushcx** [comment]
Took a read down this one, nice use of scopes. And yeah, let's get #563 merged and then this one.

File: app/models/story.rb [hunk 1]
```diff
@@ -29,6 +29,7 @@ class Story < ApplicationRecord
   scope :base, -> { includes(:tags).unmerged.not_deleted }
   scope :deleted, -> { where(is_expired: true) }
   scope :not_deleted, -> { where(is_expired: false) }
+  scope :merged, -> { where(merged_story_id: nil) }
```
[the remark this answers is not in the document]
Comment (evexoio) [reply app/models/story.rb]: It **is** confusing, I tried to add `.not` to the merged scope, but it broke the fix and I couldn't see my merged stories.

I will be Investigating

[the remark this answers is not in the document]
Comment (evexoio) [reply app/models/story.rb]: @pushcx I added PR #608 fixing the naming.

[the remark this answers is not in the document]
Comment (jmks) [reply app/models/story.rb]: Yes, I probably meant to put the `.not` in there. Turns out it is not relevant for the query I was using this scope for, so I've removed it.

**Comment by jmks** [comment]
In the `StoryRepository#newest_by_user`, the `.merged` scope not actually being used. The`where` clause immediately after it would ensure the stories where merged from the association `merged_stories`. Here's the SQL generated with and without that scope:

```
# with stories.merged scope
SELECT  `stories`.* FROM `stories` LEFT OUTER JOIN `stories` `merged_stories_stories` ON `merged_stories_stories`.`merged_story_id` = `stories`.`id` WHERE `stories`.`is_expired` = FALSE AND `stories`.`merged_story_id` IS NULL AND (`stories`.`user_id` = 12 OR `merged_stories_stories`.`user_id` = 12) ORDER BY `stories`.`id` DESC LIMIT 11

# without stories.merged scope 
SELECT  `stories`.* FROM `stories` LEFT OUTER JOIN `stories` `merged_stories_stories` ON `merged_stories_stories`.`merged_story_id` = `stories`.`id` WHERE `stories`.`is_expired` = FALSE AND (`stories`.`merged_story_id` IS NULL AND `stories`.`user_id` = 12 OR `merged_stories_stories`.`user_id` = 12) ORDER BY `stories`.`id` DESC LIMIT 11

```

They are the same up to parentheses, but the logic `1 AND 2 AND (3 OR 4)` should be the same as `1 AND (2 AND 3 OR 4)`

**Comment by pushcx** [comment]
Github's a little confused - I merged this down in the command line to fix up a style issue.

Thanks, @jmks, for this. Sorry it took so long to get merged, but I'm glad to have this fix.

**Pull request closed without merging**